gcore/src/exec.rs

```diff
@@ -220,6 +220,33 @@ pub fn gas_available() -> u64 {
     gas
 }
 
+/// Get the current amount of gas limit for execution.
```

Comment:
Incorrect comment: it's about `gas_available`
`gr_gas_limit` should return (consider for gasless sendings) amount of gas message was initially sent with

Reply:
I'd go with something like `enum InitialGas { GasFull (u64), Gasless (u64) }`

Reply:
I think the first variant is better